st3/sublime_lib/pure_resource_path.py

```diff
@@ -0,0 +1,103 @@
+import posixpath
+from functools import total_ordering
+
+
+__all__ = ['PureResourcePath']
+
+
+@total_ordering
+class PureResourcePath():
+    """A representation of a Sublime Text resource path. Inspired by pathlib."""
+
+    def __init__(self, *pathsegments):
+        self._parts = tuple(
+            part
+            for segment in pathsegments if segment
+            for part in posixpath.normpath(str(segment)).split('/')
+        )
+
+    def __hash__(self):
+        return hash(self.parts)
+
+    def __repr__(self):
+        return "%s(%s)" % (type(self).__name__, repr(str(self)))
+
+    def __str__(self):
+        return '/'.join(self.parts)
+
+    def __eq__(self, other):
+        return self._parts == other.parts
+
+    def __lt__(self, other):
+        return self._parts < other.parts
+
+    def __truediv__(self, other):
+        return self.joinpath(other)
+
+    def _new(self, *args):
+        return type(self)(*args)
+
+    @property
+    def parts(self):
+        """A tuple giving access to the path’s various components."""
+        return self._parts
+
+    @property
+    def parent(self):
+        if len(self._parts) > 1:
+            return self._new(*self._parts[:-1])
+        else:
+            return self
+
+    @property
+    def parents(self):
+        parent = self.parent
+        if self == parent:
+            return []
+        else:
+            return [parent] + parent.parents
+
+    @property
+    def name(self):
+        try:
+            return self._parts[-1]
+        except IndexError:
+            return None
```

Comment:
Should return an empty string.
```suggestion
            return ''
```
```
In [20]: PurePath("").name
Out[20]: ''
```

This property is also used in various places and expected to be a string.

Reply:
Or not?

Reply:
Done now. :-)

Reply:
Still not seeing it :confused: